types/testingutils/verifier.go

```diff
@@ -22,14 +22,11 @@ var (
 )
 
 func NewTestingVerifier() types.SignatureVerifier {
+	testingVerifierInstanceLock.Lock()
+	defer testingVerifierInstanceLock.Unlock()
 	if testingVerifierInstance == nil {
-		testingVerifierInstanceLock.Lock()
-		defer testingVerifierInstanceLock.Unlock()
-		if testingVerifierInstance == nil {
-			testingVerifierInstance = &testingVerifier{
-				signaturesCache: make(map[types.OperatorID]map[[32]byte][]byte),
-			}
-			return testingVerifierInstance
+		testingVerifierInstance = &testingVerifier{
+			signaturesCache: make(map[types.OperatorID]map[[32]byte][]byte),
```

Comment:
Can you use `go.Once()` as recommended by wikipedia?
https://en.wikipedia.org/wiki/Double-checked_locking#Usage_in_Go

if you scroll down to the java solution you see it can become tricky under certain memory models...
Before the race condition didn't always appear, I think now you solved it but let's do it the proper performant go way

Reply:
Notice that the problem is not with the singleton implementation but with the race conditions between the access in the constructor and other methods.
`go.Once` won't not solve it alone. I still need to leave the lock there.
Changing.

Reply:
Changing.